pkg/cloud/externalconn/connection.go

```diff
@@ -14,46 +14,20 @@ import (
 	"context"
 	"net/url"
 
-	"github.com/cockroachdb/cockroach/pkg/cloud"
 	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn/connectionpb"
 )
 
-// Connection is a marker interface for objects that support interaction with an
-// external resource.
-type Connection interface{}
-
-// ConnectionContext contains the dependencies passed to external connection
-// implementation during creation.
-type ConnectionContext interface {
-	// ExternalStorageContext returns the dependencies passed to external storage
-	// implementation during creation.
-	ExternalStorageContext() cloud.ExternalStorageContext
-	// KMSEnv returns the environment in which a KMS is configured and used.
-	KMSEnv() cloud.KMSEnv
-}
-
-// ConnectionDetails is the interface to the external resource represented by an
-// External Connection object.
-type ConnectionDetails interface {
-	// Dial establishes a connection to the external resource.
-	//
-	// A non-empty subdir results in a `Connection` to the joined path of the
-	// endpoint represented by the external connection and subdir.
-	Dial(ctx context.Context, connectionCtx ConnectionContext, subdir string) (Connection, error)
-	// ConnectionProto prepares the ConnectionDetails for serialization.
-	ConnectionProto() *connectionpb.ConnectionDetails
+// ExternalConnection is the interface to the external resource represented by
+// an External Connection object. This interface should expose read-only
+// methods that are required to interact with the External Connection object.
+type ExternalConnection interface {
 	// ConnectionType returns the type of the connection.
 	ConnectionType() connectionpb.ConnectionType
+	// ConnectionProto prepares the ExternalConnection for serialization.
+	ConnectionProto() *connectionpb.ConnectionDetails
```

Comment:
Maybe we shouldn't mention serialization? But mention something opposite: i.e. don't take that proto and serialize it? It's meant to be an in-memory representation...

Reply:
hmm we do serialize it thought when creating an external connection. We do:

```
ex.SetConnectionDetails(*exConn.ConnectionProto())
return ex.Create(....)
```

Reply:
But .. "we do" is different from "caller does"... Like it's okay for us to sertialize this; but don't take that proto
and stick it in e.g. changefeed job record.

Reply:
ahh okay, makes sense. Changing the comment